### PR TITLE
feat: login page

### PR DESCRIPTION
**Pull request by Dudleyneedham**
## fixes https://github.com/KILTprotocol/ticket/issues/898

- User gets password prompt
- password is salted and hashed
- result is used to decrypt data
- if not successful -> password is wrong
- use same result to encrypt future data
- error handling
- styling

## How to test:

Spin up the demo-client and try to enter the site with correct and incorrect passwor.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by Dudleyneedham** [comment]
Screenshot 2021-01-21 at 13 32 15

File: src/containers/LoginGate/LoginGate.scss [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  & .logo-id {
+    height: 100px;
+
+    & img {
+      height: 100%;
+      width: auto;
+    }
+  }
+
+  & form {
```
Comment (tjwelde) [inline src/containers/LoginGate/LoginGate.scss:67]: You should only style what is actually inside this component (LoginGate), not the other components, which are included. _Separation of Concerns_

**Review by tjwelde — CHANGES_REQUESTED** [review]

**Comment by tjwelde** [comment]
Closd button is not aligned
Screenshot 2021-01-26 at 14 11 15

**Comment by tjwelde** [comment]
"password" is written wrong and could use a question mark at the end.
Screenshot 2021-01-26 at 14 12 31

**Comment by tjwelde** [comment]
Lots of different paddings. Please make it more consistent
Screenshot 2021-01-26 at 14 13 36

**Comment by tjwelde** [comment]
Form doesn't allow 12 character passwords, besides it saying that you should use 12 characters or more

File: src/components/Register/Register.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  const [password, setPassword] = useState('')
+  const [error, setError] = useState(false)
+  const [errorMessage, setErrorMessage] = useState('')
+
+  const handleSubmit = (): void => {
+    if (!password) {
+      setErrorMessage('Please, enter a password')
+      return setError(true)
+    }
+    if (password.length <= 11) {
```
Comment (tjwelde) [inline src/components/Register/Register.tsx:18]: Why not just
```suggestion
    if (password.length < 12) {
```
?

File: src/components/Register/Register.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            <div>
+              <input
+                value={password}
+                type="password"
+                onChange={e => setPassword(e.target.value)}
+              />
+            </div>
+          </label>
+          <button type="submit" className="confirm">
+            Login
```
Comment (tjwelde) [inline src/components/Register/Register.tsx:56]: Rename to "Register", or "Create Account"

**Review by tjwelde — CHANGES_REQUESTED** [review]
The input and the button are not horizontally aligned (both login and register)
Screenshot 2021-01-26 at 18 23 45

The close button is still not aligned well. it has more margin on bottom than on the top. Should be in the middle vertically.
Screenshot 2021-01-26 at 18 25 05

File: src/components/Register/Register.tsx (hunk 3 above)
Comment (Dudleyneedham) [reply src/components/Register/Register.tsx:56]: Changed to "Create Account".

**Comment by Dudleyneedham** [comment]
> The input and the button are not horizontally aligned (both login and register)
> Screenshot 2021-01-26 at 18 23 45
> 
> The close button is still not aligned well. it has more margin on bottom than on the top. Should be in the middle vertically.
> Screenshot 2021-01-26 at 18 25 05

I have updated it through-out the whole application. I know it isn't part of the ticket, but the issue is from the `Modal.scss` so I fixed it there.

Let me know how you find it? 
Screenshot 2021-01-27 at 10 40 08

**Comment by tjwelde** [comment]
I like it

**Review by tjwelde — APPROVED** [review]
LGTM

**Pull request merged**